include/dl/dl.h

```diff
@@ -411,6 +411,70 @@ typedef struct dl_instance_info
 */
 dl_error_t DL_DLL_EXPORT dl_instance_get_info( const unsigned char* packed_instance, size_t packed_instance_size, dl_instance_info_t* out_info );
 
+/*
+	Function: dl_get_metadata_cnt
+	    Fetch the number of meta data instances for the specified type, enum or union
+
+	Parameters:
+	    dl_ctx          - Context to load type-library into.
+	    type            - Type id for type to fetch the meta data count of.
+	    count           - The number of meta data instances available for the give type
+
+	Return:
+	    DL_ERROR_OK on success.
+*/
+dl_error_t DL_DLL_EXPORT dl_get_metadata_cnt( dl_ctx_t dl_ctx, dl_typeid_t type, unsigned int* count );
+
+/*
+	Function: dl_get_member_metadata_cnt
+	    Fetch the number of meta data instances for the specified type member, enumerator or union member
+
+	Parameters:
+	    dl_ctx          - Context to load type-library into.
+	    type            - Type id for type to fetch the meta data count of.
+	    item_index      - The index of the member, enumerator or union member
+	    count           - The number of meta data instances available for the give type
+
+	Return:
+	    DL_ERROR_OK on success.
+*/
+dl_error_t DL_DLL_EXPORT dl_get_member_metadata_cnt( dl_ctx_t dl_ctx, dl_typeid_t type, unsigned int item_index, unsigned int* count );
+
+/*
+	Function: dl_get_metadata
+		Fetch the number of meta data instances for the specified type
+
+	Parameters:
+		dl_ctx            - Context to load type-library into.
+		type              - Type id for type to fetch the meta data instance of.
+		index             - The index of the meta data to fetch
+		out_type_id       - The type id of the returned meta data instance
+	    out_instance_size - The size of the returned instance
+		out_instance      - A pointer to the instance
+
+	Return:
+		DL_ERROR_OK on success.
+*/
+dl_error_t DL_DLL_EXPORT dl_get_metadata( dl_ctx_t dl_ctx, dl_typeid_t type, unsigned int index, dl_typeid_t* out_type_id, unsigned int* out_instance_size, void** out_instance );
```

Comment:
I guess I will find out when I read on in the CL, but it is at least unclear from the doc. Do I get data that need to be manually patched from this.
Is there any reason not to just get a void* to an already loaded and patched instance? I.e. what would I want the size for?

Reply:
Hmm, the data is patched. And the size is not meaningful, good catch!